File: Programacion 4/Euler Primos/euler.py
def calcularEuler():
    const = 10**2000
    e = 0
    f = 1
    for i in range(0,2000):
        if(i != 0):
            f = f * i
        k = const // f
        e += k
    return e
def eratostenes(Q):
    sieve= [False, False] + [True for i in range(2,Q)]
    for i in range(2,Q):
        if sieve[i]==True:
            j = 2*i
            while j<Q:
                sieve[j] = False
                j = j+i
    return sieve
if __name__ == "__main__":
    first = 0
    mayor = 0
    sieve = eratostenes(1000000)
    e = calcularEuler()
    cad = str(e)
    euler = [int(cad[i]) for i in range(1000)]
    numeros = []
    for i in range(0,995):
        x=0
        for j in range(0,6):
            x = x + (euler[i+j]*(10**(5-j)))
        if x>100000 and sieve[x]==True:
            if first==0:
                first = x
            if x > mayor:
                mayor = x
    print("primero = ", first)
    print("mayor = ", mayor)

File: Programacion 4/Euler Primos/Euler.java
import java.util.*;
import java.math.*;

public class Euler{
	public static BigDecimal calculoeEuler(){
		BigDecimal numE = new BigDecimal(2);
		BigDecimal f = new BigDecimal(1);
		for(int i=2; i<1000; i++){
			f = f.multiply(new BigDecimal(i));
			BigDecimal k = (new BigDecimal(1)).divide(f, 2000, RoundingMode.HALF_DOWN);
			numE = numE.add(k);
		}
		return(numE);
	}

	public static void Eratostenes(int lim, int[] sieve){
		int i,j;
		sieve[0]=0; sieve[1]=0;
		for(i=2; i<lim; i++){
			sieve[i]=1;
		}
		for(i=2; i<lim; i++){
			if(sieve[i]==1){
				for(j=2*i; j<lim; j+=i){
					sieve[j]=0;
				}
			}
		}
	}

	public static void main(String[] args){
		int x,i,j,lim = 1000000, mayor, first = 0;
		int sieve[] = new int[1000000];
		Eratostenes(lim, sieve);
		BigDecimal numE = calculoeEuler();
		String cad = numE.toPlainString();
		cad = cad.replace(".","");
		cad = cad.substring(0,1000);
		for(i=0,mayor=0; i<=994; i++){
			x=0;
			for(j=0; j<6; j++){
				x+= (cad.charAt(i+j)-'0')* Math.pow(10,6-j-1);
			}
			if(sieve[x]==1){
				if(first==0)
				first= x;
				if(x>mayor)
				mayor=x;
			}
		}
		System.out.println("primero = " + first);
		System.out.println("mayor = " + mayor);
	}
}